include/device.h

```diff
@@ -568,6 +568,33 @@ int device_required_foreach(const struct device *dev,
  */
 __syscall const struct device *device_get_binding(const char *name);
 
+/**
+ * @brief Retrieve the device structure for a driver by name.
+ *
+ * This is the implementation underlying device_from_name(), without
+ * the overhead of a syscall wrapper.
+ *
+ * @param name device name to search for.  A null pointer, or a pointer to an
+ * empty string, will cause NULL to be returned.
+ *
+ * @return pointer to device structure; NULL if the name is not
+ * acceptable or no device is associated with the name.
+ */
+const struct device *z_device_from_name(const char *name);
```

Comment:
where is this being used?

Reply:
This is provided so driver init functions and other code already known to be running in privileged mode doesn't have to go through a system call.

Reply:
But it looks like there is no implementation of this function. I think you need to add something like:
```
const struct device *z_device_from_name(const char *name)
{
	return device_lookup_by_name(name, false);
}
```
in `device.c`.

Reply:
Yes, you're right.